consensus/spos/bls/subroundEndRound.go

```diff
@@ -621,7 +672,9 @@ func (sr *subroundEndRound) getFullMessagesForInvalidSigners(invalidPubKeys []st
 }
 
 func (sr *subroundEndRound) handleInvalidSignersOnAggSigFail() ([]byte, []byte, error) {
-	invalidPubKeys, err := sr.verifyNodesOnAggSigFail()
+	ctx, cancel := context.WithTimeout(context.Background(), sr.RoundHandler().TimeDuration())
```

Comment:
in case of timeout, we need to make sure that the goroutines are not panicking, check my previous comment.